CMSgov/mint-app
# [EASI-2267] - Prepare for clearance

# EASI-2267

## Changes and Description

- Prepare for clearance task list item
- Checklist form component
- Review (readonly) component
- Subscription lock/modal on review component
- Clearance warning/modal if already cleared
- Unit/cypress testing

Out of scope but made ticket much easier to develop
- Context provider for getting model plan info from anywhere in app


![Screen Shot 2022-10-25 at 9 56 53 AM](https://user-images.githubusercontent.com/95709965/197793190-95085af4-9910-424d-83ba-72e662eac991.png)

![Screen Shot 2022-10-25 at 9 57 13 AM](https://user-images.githubusercontent.com/95709965/197793292-265da8e9-b1ce-4c41-afc1-d77edff16900.png)

![Screen Shot 2022-10-25 at 9 57 38 AM](https://user-images.githubusercontent.com/95709965/197793380-ae7aa3b3-161e-4565-b6b2-a6e43e6c34f9.png)

![Screen Shot 2022-10-25 at 9 57 57 AM](https://user-images.githubusercontent.com/95709965/197793459-5eabeafc-f73a-4083-b5fa-2537e9d82257.png)

![Screen Shot 2022-10-25 at 9 58 14 AM](https://user-images.githubusercontent.com/95709965/197793526-820f7b12-15d7-410c-9403-29467d435685.png)

## How to test this change

- Seed the database
- Enter into `Empty Plan` and verify the info box on `Prepare for clearance` task list section
- Enter into the `Basics` and populate the `clearanceStarts` with a date within 20 days
- Return to task list and verify `Prepare for clearance` is now `Ready`
- Enter into the task list item and check/uncheck checkboxes and update status
- Verify task list status is now `In progress`
- From the checklist, review all sections and try to edit a section that has already been cleared
- Confirm modal when trying to access cleared section
- GQL subscription lock modal will appear if another browser open and occupying task list section

## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
